keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryArtifactRepository.kt

```diff
@@ -169,6 +172,14 @@ class InMemoryArtifactRepository(
       .filter {
         if (artifact is DebianArtifact && artifact.statuses.isNotEmpty()) {
           it.status in artifact.statuses
+        } else if (artifact is DockerArtifact) {
+          // we only want to give valid versions, so this will filter out tags like "latest"
+          // and others that don't fit the chosen versioning strategy.
+          try {
+            it.version != "latest" && TagComparator.parseWithRegex(it.version, artifact.tagVersionStrategy, artifact.captureGroupRegex) != null
```

Comment:
Wouldn’t “latest” be filtered out naturally as a result of the regexp not matching? Why check for it explicitly?

Reply:
Ha! You would think! And I did think that! But, it turns out, no. With some tag strategies you say "hey I promise I'll give you a real tag that's a number" and that regex looks like `(^.*$)`. And that regex just means "grab everything". We could make the regex more complicated, to filter out other things. And a user could totally do that also. But for now, I thought I'd just filter out latest because I know we definitely don't want that ever.